### PR TITLE
fix(sbb-tag): fix accessibility by using buttons instead of checkboxes

### LINKED ISSUE #1594: [A11y] Tags check focus
## Detailed description
According to our first accessibility review by Mürsel Genc, the focus state for the underlying checkboxes of the tag component is assigned to the first element after activating a button/tag.

It would be better if the focus remains on the activated element.

image

Further information: https://flow.sbb.ch/browse/SBBCH-15188

### LINKED ISSUE #1596: [A11y] Tags in high contrast mode
## Detailed description

According to our first accessibility review by Mürsel Genc, the tags are not looking good in high contrast mode. Since they are implemented as checkboxes, their label is yellow, but yellow indicates normal text in high contrast mode. The border turns green for the active/checked tags, but is yellow for unchecked tags. Please improve accessibility by turning the border white instead of yellow.

image

Further information: https://flow.sbb.ch/browse/SBBCH-15182

### LINKED ISSUE #1616: story(sbb-tag-group): refactor with button accessibility pattern
### Feature Description

We would like to refactor the sbb-tag-group implementation to use buttons with aria-pressed.
See e.g. https://material.angular.io/components/button-toggle/overview

This also should allow us to provide both "checkbox" and "radio button" use cases, via the property `multiple` on `<sbb-tag-group>`.

This should solve #1594 and #1596.

### Definition of Done

- [ ] Component is implemented
- [ ] Documentation is provided
- [ ] Tests are implemented
- [ ] E2E Tests are implemented
- [ ] Storybook stories are implemented
- [ ] Navigation via keyboard is tested
- [ ] Screen reader output is tested
- [ ] High-contrast is tested
- [ ] Remaining accessibility is tested
- [ ] UX approved

### PR DESCRIPTION
**Pull request by jeripeierSBB**
## Issue

This PR Closes #1616
This PR Closes #1594
This PR Closes #1596

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- Use buttons instead of checkboxes as underlaying pattern

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [ ] Edge Desktop
- [x] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [x] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [x] NVDA Chrome Desktop
- [x] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

### CODE REVIEW

File: src/components/sbb-tag/sbb-tag.tsx [hunk 1]
```diff
@@ -55,56 +59,90 @@ export class SbbTag implements ComponentInterface, AccessibilityProperties {
...
-      <label class="sbb-tag__wrapper">
-        <input
-          ref={(checkbox: HTMLInputElement) => (this._checkbox = checkbox)}
-          type="checkbox"
+      <Host role="presentation">
+        <button
+          type="button"
+          class="sbb-tag"
+          // We have to ensure that the value is always present
+          aria-pressed={this.checked ? 'true' : 'false'}
```
Comment (DavideMininni-Fincons) [inline src/components/sbb-tag/sbb-tag.tsx:139]:
```suggestion
          aria-pressed={String(this.checked)}
```
could it work?

**Review by DavideMininni-Fincons — COMMENTED** [review]
Looks good, I added just a couple of comment :)

File: src/components/sbb-tag-group/sbb-tag-group.tsx [hunk 2]
```diff
@@ -13,43 +22,116 @@ import { isValidAttribute } from '../../global/helpers/is-valid-attribute';
...
+      this.value = event.detail.value;
+      // If a value was checked in exclusive mode, assign this value directly.
+    } else if (event.detail.type === 'checked' && event.detail.checked) {
+      this.value = target.value;
+    }
+  }
+
+  private _updateValueByReadingTags(): void {
+    if (this.multiple) {
+      const values = [];
```
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/components/sbb-tag-group/sbb-tag-group.tsx:101]: From readability perspective I'd implemented like you suggested (nicer to read! :) ), but from performance perspective it has to loop two times through all (or at least filtered items), so we once decided in these cases to switch to a for-loop.

File: src/components/sbb-tag-group/sbb-tag-group.e2e.ts [hunk 3]
```diff
@@ -3,37 +3,576 @@ import { E2EElement, E2EPage, newE2EPage } from '@stencil/core/testing';
...
+
+        await page.waitForChanges();
+
+        expect(await element.getProperty('value')).toEqual('tag2');
+        expect(await element.find('sbb-tag#sbb-tag-1')).not.toHaveAttribute('checked');
+        expect(await element.find('sbb-tag#sbb-tag-2')).toHaveAttribute('checked');
+        expect(await element.find('sbb-tag#sbb-tag-3')).not.toHaveAttribute('checked');
+      });
+
+      it('should ignore because value is an array', async () => {
```
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/components/sbb-tag-group/sbb-tag-group.e2e.ts:496]: In exclusive mode, it should always ignore an array, regardless if it is empty or not, so I'd leave the original title?

File: src/components/sbb-tag/sbb-tag.tsx (hunk 1 above)
Comment (jeripeierSBB) [reply src/components/sbb-tag/sbb-tag.tsx]: thanks! changed to toString() as found in other examples in lyne. But your suggestion would work as well :)

File: src/components/sbb-tag-group/sbb-tag-group.e2e.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (DavideMininni-Fincons) [reply src/components/sbb-tag-group/sbb-tag-group.e2e.ts:496]: No, it's fine

**Review by dauriamarco — APPROVED** [review]
LGTM 👍

File: src/components/sbb-tag-group/sbb-tag-group.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/components/sbb-tag-group/sbb-tag-group.tsx:101]: I think I echo Davides sentiment. I think in this scenerio where we don't expect a large amount of tags, we should be "safe" with the suggestion from Davide.

File: src/components/sbb-tag-group/sbb-tag-group.tsx [hunk 4]
```diff
@@ -13,43 +22,116 @@ import { isValidAttribute } from '../../global/helpers/is-valid-attribute';
...
-  private _onTagSlotChange(): void {
-    this.tags = this._getTags();
+  private _ensureOnlyOneTagSelected(): void {
+    if (this.multiple) {
+      return;
+    }
+
+    // Ensure only one tag checked
+    const checkedTags = this._tags().filter((tag) => tag.checked);
+    if (checkedTags.length > 1) {
```
Comment (kyubisation) [inline src/components/sbb-tag-group/sbb-tag-group.tsx:77]: Nitpick; The if check is not strictly necessary, as you could just call slice(1)

**Review by kyubisation — APPROVED** [review]
LGTM 👍 
Some nitpicks 😀

File: src/components/sbb-tag-group/sbb-tag-group.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/components/sbb-tag-group/sbb-tag-group.tsx:101]: Nice ;)

**Pull request merged**